### DISCUSSION TITLE
Triggering new layer with template content

### DISCUSSION THREAD

**Discussion by davisums** [unpoly/unpoly#673] [General]
I'm trying to trigger a new layer, using the content of a template that is already on the page. I found this snippet in another discussion:

```
up.on('up:link:follow', 'a[up-layer=new]', function({renderOptions}) {
  if (renderOptions.url.startsWith('#')) {
    // Find the <template> with an ID matching the given URL.
    // E.g. if we're given { url: '#foo' } we will look for an element "template#foo".
    let template = document.querySelector("template" + renderOptions.url)
    renderOptions.url = undefined
    renderOptions.content = template.innerHTML
  }
})
```

The link looks like this:
`<a id="newsletter-signup-popup-link" up-href="#newsletter-signup-popup" up-mode="popup" up-layer="new" up-follow>open hier</a>`

This works fine for when a link is clicked, but I'm trying to trigger this programatically using JS.
```
       let link = document.getElementById('newsletter-signup-popup-link');
        up.follow(link);
```

What I'm currently doing is hiding that link through css, and then trigger it. Is there any way to not have this link in the DOM at all?

Also, any plans to add something like this natively? I think for small popups in particular (like context menus)  it makes a lot of sense.

**Comment by triskweline** [comment]
> Is there any way to not have this link in the DOM at all?

You could call `up.render({ content: document.getElementById('newsletter-signup-popup-link').innerHTML })`.

> Also, any plans to add something like this natively? I think for small popups in particular (like context menus) it makes a lot of sense.

I agree!
